src/controllers/auth.controller.js

```diff
@@ -0,0 +1,141 @@
+
+function validateEmail(value) {
+  if (!value) {
+    return 'Email is required';
+  }
+
+  const emailPattern = /^[\w.+-]+@([\w-]+\.){1,3}[\w-]{2,}$/;
+
+  if (!emailPattern.test(value)) {
+    return 'Email is not valid';
+  }
+}
+
+function validatePassword(value) {
+  if (!value) {
+    return 'Password is required';
+  }
+
+  if (value.length < 6) {
+    return 'At least 6 characters';
+  }
+}
+
+async function generateTokens(res, user) {
+  const normalizedUser = userService.normalize(user);
+  const accessToken = jwtService.sign(normalizedUser);
+  const refreshToken = jwtService.signRefresh(normalizedUser);
+
+  await toeknService.save(normalizedUser.id, refreshToken);
+
+  res.cookie('refreshToken', refreshToken, {
+    maxAge: 30 * 24 * 60 * 60 * 1000,
+    HttpOnly: true,
+  });
+  res.send({
+    user: normalizedUser,
+    accessToken
+  });
+}
+
+const register = async (req, res) => {
+  const { email, password } = req.body;
+
+  const errors = {
+    email: validateEmail(email),
+    password: validatePassword(password),
+  };
+
+  if (errors.email || errors.password) {
+    throw ApiError.badRequest('Bad request', errors);
+  }
+
+  const hashedPassword = await bcrypt.hash(password, 10); // problem
+
+  await userService.register(email, hashedPassword);
+
+  res.send({ message: 'Okay' });
+};
+
+const activate = async (req, res) => {
+  const { activationToken } = req.params;
+  const user = await User.findOne({ where: { activationToken }});
+
+  if (!user) {
+    return res.status(404).send('No such user');
+  }
+
+  user.activationToken = null;
+  user.save();
+
+  res.send(user);
+};
+
+const login = async (req, res) => {
+  const { email, password } = req.body;
+
+  const user = await userService.findByEmail(email);
+
+  if (!user) {
+    throw ApiError.badRequest('No such user');
+  }
+
+  const isUserValide = await bcrypt.compare(password, user.password);
+
+  if (!isUserValide) {
+    throw ApiError.badRequest('Incorrect password');
+  }
+
+  generateTokens(res, user);
+
+  // const normalizedUser = userService.normalize(user);
+  // const accessToken = jwtService.sign(normalizedUser);
+
+  // res.send({
+  //   user: normalizedUser,
+  //   accessToken
+  // });
+};
+
+const refresh = async (req, res) => {
+  const { refreshToken } = req.cookies;
+
+  const userData = jwtService.verifyRefresh(refreshToken);
+  const token = await toeknService.getByToken(refreshToken);
+
+  if (!userData || !token) {
+    throw ApiError.unAuthorized();
+    return;
+  }
+
+  const user = await userService.findByEmail(userData.email);
+  generateTokens(res, user);
+};
+
+const logout = async (req, res) => {
+  const { refreshToken } = req.cookies;
+  const userData = jwtService.verifyRefresh(refreshToken);
+
+  if (!userData || !refreshToken) {
+    throw ApiError.unAuthorized();
+    return;
+  }
+
+  await toeknService.remove(userData.id);
+
+  res.status(200).send('You were succesfully logout');
```

Comment:
There is a typo in the response message. It should be 'You were successfully logged out'.